airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/steps/gradle.py

```diff
@@ -32,210 +36,129 @@ class GradleTask(Step, ABC):
 
-        yum_packages_to_install = [
-            "docker",  # required by :integrationTestJava.
-            "findutils",  # gradle requires xargs, which is shipped in findutils.
-            "jq",  # required by :acceptance-test-harness to inspect docker images.
-            "rsync",  # required for gradle cache synchronization.
-        ]
+    @property
+    def gradle_command(self):
+        connector_gradle_task = f":airbyte-integrations:connectors:{self.context.connector.technical_name}:{self.gradle_task_name}"
+        return self._get_gradle_command(connector_gradle_task)
 
-        # Common base container.
-        gradle_container_base = (
-            self.dagger_client.container()
-            # Use a linux+jdk base image with long-term support, such as amazoncorretto.
-            .from_(AMAZONCORRETTO_IMAGE)
-            # Mount the dependency cache volume, but not to $GRADLE_HOME, because gradle doesn't expect concurrent modifications.
-            .with_mounted_cache(self.GRADLE_DEP_CACHE_PATH, self.dependency_cache_volume, sharing=CacheSharingMode.LOCKED)
-            # Set GRADLE_HOME to the directory which will be rsync-ed with the gradle cache volume.
-            .with_env_variable("GRADLE_HOME", self.GRADLE_HOME_PATH)
-            # Same for GRADLE_USER_HOME.
-            .with_env_variable("GRADLE_USER_HOME", self.GRADLE_HOME_PATH)
-            # Install a bunch of packages as early as possible.
-            .with_exec(
-                sh_dash_c(
-                    [
-                        # Update first, but in the same .with_exec step as the package installation.
-                        # Otherwise, we risk caching stale package URLs.
-                        "yum update -y",
-                        f"yum install -y {' '.join(yum_packages_to_install)}",
-                        # Remove any dangly bits.
-                        "yum clean all",
-                        # Deliberately soft-remove docker, so that the `docker` CLI is unavailable by default.
-                        # This is a defensive choice to enforce the expectation that, as a general rule, gradle tasks do not rely on docker.
-                        "yum remove -y --noautoremove docker",  # remove docker package but not its dependencies
-                        "yum install -y --downloadonly docker",  # have docker package in place for quick install
-                    ]
-                )
+    @contextmanager
+    def gradle_environment(self):
+        """
+        Context manager to set the gradle environment:
+        - Check if the system has all the required commands in the path.
+        - Set the S3 build cache environment variables if available.
+        - ... Add whatever setup/teardown logic needed to run a gradle task.
+        """
+        secrets_written = []
+        try:
+            # Check if the system has all the required commands in the path.
+            self.check_system_requirements()
```

Comment:
making sure I understand: this is required to propagate the various gradle caching to the test subprocess? It's not automatically inherited?

Reply:
That's just a safety net to check that the dev / runner has all the required dependencies installed on their system. It will crash the pipeline with an explicit error instead of tryng to debug a less explicit gradle failure.

Reply:
I think your question related to the `gradle_environment` method. 
> is is required to propagate the various gradle caching to the test subprocess? It's not automatically inherited?

ATM the s3 creds can be passed either as a CLI option or as an env var and are then available as secrets in the context object. I prefer to not assume that the env var is already set and don't see an issue in overwriting it.

Reply:
got it. I was reading too fast and assumed we are enforcing the presence of these S3 env vars - looks this is already optional, nice!